pkg/entrypoint/vhd-footer/vhd-footer.go

```diff
@@ -0,0 +1,104 @@
+package vhdfooter
+
+import (
+	"bytes"
+	"encoding/binary"
+	"flag"
+	"os"
+	"time"
+
+	uuid "github.com/satori/go.uuid"
+	"github.com/sirupsen/logrus"
+
+	"github.com/openshift/openshift-azure/pkg/util/log"
+)
+
+var (
+	size = flag.Uint64("size", 0, "image size in bytes")
+)
+
+// from
+// http://download.microsoft.com/download/f/f/e/ffef50a5-07dd-4cf8-aaa3-442c0673a029/Virtual%20Hard%20Disk%20Format%20Spec_10_18_06.doc
+
+type vhdFooter struct {
+	Cookie             [8]byte
+	Features           uint32
+	FileFormatVersion  uint32
+	DataOffset         uint64
+	TimeStamp          uint32
+	CreatorApplication [4]byte
+	CreatorVersion     uint32
+	CreatorHostOS      [4]byte
+	OriginalSize       uint64
+	CurrentSize        uint64
+	DiskGeometry       uint32
+	DiskType           uint32
+	Checksum           uint32
+	UniqueID           uuid.UUID
+	SavedState         uint8
+	Reserved           [427]byte
+}
+
+func (f *vhdFooter) CalculateChecksum() error {
+	f.Checksum = 0
+
+	buf := &bytes.Buffer{}
+
+	err := binary.Write(buf, binary.BigEndian, f)
+	if err != nil {
+		return err
+	}
+
+	for _, b := range buf.Bytes() {
+		f.Checksum += uint32(b)
+	}
+	f.Checksum = ^f.Checksum
+
+	return nil
+}
+
+func main() {
```

Comment:
2 `main()` functions? something is off here. + `start()` does not do anything. Please review again